prow/cmd/peribolos/main.go

```diff
@@ -146,13 +146,13 @@ func main() {
 
 	o := parseOptions()
 
-	githubClient, err := o.github.GitHubClient(!o.confirm)
+	githubClient, err := o.github.GitHubClient((o.github.AppID == "") && !o.confirm)
```

Comment:
Here we're defining dry-run as "not using GitHub App auth and not asking for `--confirm`", right? So passing `--github-app-id=something --confirm` won't actually honor the `--confirm` setting? We might want to simplify this or return an error or something? To me asking for `--confirm` but not getting it silently seems surprising.

Reply:
If you pass `--dump some_org --github-app-id=something --confirm` it will honor the `--confirm` because it will not run as a dry-run and the config will be dumped successfully. But if you pass `--dump some_org --github-app-id=something` this will silently add `--confirm` so that peribolos doesn't run in dry-run mode and an installation token can be created (peribolos throws an error if it tries to create an instllation token in dry-run mode), and I agree with you that it shouldn't.

In the pushed fix I tried to simplify it a bit and I put the added dry-run logic to the argument parser. This way if we pass `--dump some_org --github-app-id=something --confirm` it will work fine. If we pass `--dump some_org --github-app-id=something` it will throw out an error saying with these arguments it can't run in dry-run mode. 

This change doesn't affect peribolos when just a token is used with `--dump` option, in this mode, it still can't run with the `--confirm` option.